flink-table/flink-sql-gateway/src/main/java/org/apache/flink/table/gateway/service/materializedtable/MaterializedTableManager.java

```diff
@@ -414,6 +422,78 @@ protected static String getManuallyRefreshStatement(
         return insertStatement.toString();
     }
 
+    private static ResultFetcher callDropMaterializedTableOperation(
+            OperationExecutor operationExecutor,
+            OperationHandle handle,
+            DropMaterializedTableOperation dropMaterializedTableOperation) {
+        ObjectIdentifier tableIdentifier = dropMaterializedTableOperation.getTableIdentifier();
+        CatalogMaterializedTable materializedTable =
+                getCatalogMaterializedTable(operationExecutor, tableIdentifier);
+
+        if (CatalogMaterializedTable.RefreshStatus.ACTIVATED
```

Comment:
We should also consider the `INTIALIZING` status, the refresh may be submitted, we also need to handle it.

For the `INTITIALIZING` status, due to we don't known the jobId, so one simple solution is throw exception directly, we should wait the is submitted only then can we drop the materialized table. WDYT?